vernac/library.ml

```diff
@@ -436,14 +440,36 @@ let error_recursively_dependent_library dir =
 (* Security weakness: file might have been changed on disk between
    writing the content and computing the checksum... *)
 
-let save_library_base f sum lib univs tasks proofs =
+type opaques =
+| OpaqueDirect of seg_proofs
+| OpaqueAncient of Opaqueproof.opaque_proofterm Ancient.t option array
+
+let save_library_base f sum lib univs tasks (proofs : opaques) =
   let ch = raw_extern_library f in
   try
     ObjFile.marshal_out_segment ch ~segment:"summary" (sum    : seg_sum);
     ObjFile.marshal_out_segment ch ~segment:"library" (lib    : seg_lib);
     ObjFile.marshal_out_segment ch ~segment:"universes" (univs  : seg_univ option);
     ObjFile.marshal_out_segment ch ~segment:"tasks" (tasks  : 'tasks option);
-    ObjFile.marshal_out_segment ch ~segment:"opaques" (proofs : seg_proofs);
+    let () = match proofs with
+    | OpaqueDirect proofs ->
+      let iter i proof =
+        let segment = Printf.sprintf "opaques/%i" i in
+        ObjFile.marshal_out_segment ch ~segment proof
+      in
+      Int.Map.iter iter proofs
+    | OpaqueAncient proofs ->
+      let iter i = function
+      | None -> ()
+      | Some proof ->
+        let segment = Printf.sprintf "opaques/%i" i in
+        let ch, final = ObjFile.marshal_out_binary ch ~segment in
+        let () = Ancient.serialize proof ch in
+        let () = final () in
+        Ancient.delete proof
```

Comment:
If this is the only occurrence of `delete` I suggest file deletion is taken care by `serialize`.